azurerm_storage_table_entity breaks if unable to find entity in Azure but in State

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.13.4

### AzureRM Provider Version

4.51.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_table_entity

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "azureexample"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                     = "azureexamplestorage1"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_storage_table" "example" {
  name                 = "myexampletable"
  storage_account_name = azurerm_storage_account.example.name
}

resource "azurerm_storage_table_entity" "example" {
  storage_table_id = azurerm_storage_table.example.id

  partition_key = "examplepartition"
  row_key       = "examplerow"

  entity = {
    example = "example"
  }
}
```

### Debug Output/Panic Output

```shell
2025-11-06T11:24:12.186+1030 [DEBUG] provider.terraform-provider-azurerm_v4.51.0_x5: [DEBUG] az-cli invocation: az account get-access-token --scope https://examplestorage.table.core.windows.net/.default
2025-11-06T11:24:12.911+1030 [DEBUG] provider.terraform-provider-azurerm_v4.51.0_x5: [DEBUG] GET https://examplestorage.table.core.windows.net/myexampletable%28PartitionKey=%27examplepartition%27,%20RowKey=%27examplerow%27%29
2025-11-06T11:24:13.090+1030 [ERROR] provider.terraform-provider-azurerm_v4.51.0_x5: Response contains error diagnostic: diagnostic_summary="checking for existing Entity (Partition Key \"examplepartition\" / Row Key \"examplerow\" / Table Name \"myexampletable\" / Account \"Account \\\"examplestorage\\\" (IsEdgeZone false / ZoneName \\\"\\\" / Subdomain Type \\\"table\\\" / DomainSuffix \\\"core.windows.net\\\")\"): executing request: unexpected status 403 (403 Forbidden) with EOF"
```

### Expected Behaviour

If you manually delete the entity from the azure portal, Terraform fails to run as it can not find the entity to match the columns, rather than realising the entity does not exist at all.

### Actual Behaviour

╷
│ Error: checking for existing Entity (Partition Key "examplepartition" / Row Key "examplerow" / Table Name "myexampletable" / Account "Account \"azureexamplestorage1\" (IsEdgeZone false / ZoneName \"\" / Subdomain Type \"table\" / DomainSuffix \"core.windows.net\")"): executing request: unexpected status 403 (403 Forbidden) with EOF
│
│   with azurerm_storage_table_entity.example,
│   on main.tf line 24, in resource "azurerm_storage_table_entity" "example":
│   24: resource "azurerm_storage_table_entity" "example" {
│
╵

### Steps to Reproduce

Create Storage Account, Table, Entity.
Delete the Entity from the portal and then try to Apply again.

If you then remove the entity from state using terraform state rm, it all works fine

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/storage/storage_table_entity_resource.go
+++ b/internal/services/storage/storage_table_entity_resource.go
@@ -227,6 +227,10 @@ func resourceStorageTableEntityRead(d *pluginsdk.ResourceData, meta interface{})
 
 	result, err := client.Get(ctx, id.TableName, input)
 	if err != nil {
+		if response.WasNotFound(result.HttpResponse) {
+			d.SetId("")
+			return nil
+		}
 		return fmt.Errorf("retrieving %s: %v", id, err)
 	}
 

